google-ads/src/main/java/com/google/ads/googleads/lib/utils/FieldMasks.java

```diff
@@ -119,6 +124,58 @@ private static void compare(
     }
   }
 
+  /**
+   * Gets the field value of a message from a field mask path.
+   *
+   * @param fieldMaskPath The field mask path.
+   * @param entity The entity to retrieve values from.
+   * @return the field Object.
+   */
+  public static <V> V getFieldValue(String fieldMaskPath, Message entity) {
+    Deque<String> fieldMaskParts = new LinkedList<>(Arrays.asList(fieldMaskPath.split("\\.")));
+    Message currentEntity = entity;
+    while (!fieldMaskParts.isEmpty()) {
+      String fieldMaskPart = fieldMaskParts.remove();
+      Preconditions.checkNotNull(
+          currentEntity, String.format("Cannot get field value. %s is null.", fieldMaskPart));
+      Descriptor descriptor = currentEntity.getDescriptorForType();
+      FieldDescriptor childField = descriptor.findFieldByName(fieldMaskPart);
+      V childValue;
+      try {
+        childValue = (V) currentEntity.getField(childField);
+      } catch (NullPointerException e) {
```

Comment:
You'll never get to the if statement here because `descriptor.findFieldByName(fieldMaskPart)` throws a `NullPointerException` if `childField` is `null` which is why this is structured the way it is.

Reply:
![Screenshot from 2020-12-11 14-57-45](https://user-images.githubusercontent.com/410275/101918517-5a05f980-3bc1-11eb-9c39-d11b049ee2c3.png)

Looks like findFieldByName() returns null if the field doesn't exist.